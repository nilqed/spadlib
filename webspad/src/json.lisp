(in-package :webspad)


(defun bool-to-str (s) (if s "true" "false"))

(defun encode-json (data)
  (setf flags (webspad-data-format-flags data))
  (format nil "{ \"input\":~S,~
                 \"stdout\":~S,~
                 \"stderr\":~S,~
                 \"multiline?\":~S,~
                 \"spad-type\":~S,~
                 \"algebra\":~S,~
                 \"charybdis\":~S,~
                 \"tex\":~S,~
                 \"html\":~S,~
                 \"mathml\":~S,~
                 \"formula\":~S,~
                 \"fortran\":~S,~
                 \"texmacs\":~S,~
                 \"openmath\":~S,~
                 \"format-flags\": {~
                   \"algebra\":~S,~
                   \"tex\":~S,~
                   \"html\":~S,~
                   \"mathml\":~S,~
                   \"formula\":~S,~
                   \"fortran\":~S,~
                   \"texmcas\":~S,~
                   \"openmath\":~S~
                   }}"
                   (webspad-data-input data)
                   (webspad-data-stdout data)
                   (webspad-data-stderr data)
                   (bool-to-str (webspad-data-multiline? data))
                   (webspad-data-spad-type data)
                   (webspad-data-algebra data)
                   (webspad-data-charybdis data)
                   (webspad-data-tex data)
                   (webspad-data-html data)
                   (webspad-data-mathml data)
                   (webspad-data-formula data)
                   (webspad-data-fortran data)
                   (webspad-data-texmacs data)
                   (webspad-data-openmath data)
                   (bool-to-str (ws-format-algebra flags))
                   (bool-to-str (ws-format-tex flags))
                   (bool-to-str (ws-format-html flags))
                   (bool-to-str (ws-format-mathml flags))
                   (bool-to-str (ws-format-formula flags))
                   (bool-to-str (ws-format-fortran flags))
                   (bool-to-str (ws-format-texmacs flags))
                   (bool-to-str (ws-format-openmath flags))))
                   
                   
                   
                   
                   
                    
