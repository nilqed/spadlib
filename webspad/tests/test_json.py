#!/usr/bin/env python
# -*- coding: UTF-8 -*-

import requests
import json

url = 'http://localhost:4242/json'
payload = {'code': 'D(x^n,x,4)'}

r = requests.post(url, data=payload)

print(r.text)

data = r.text
data = data.replace('\r','\\r')
data = data.replace('\n','\\n')
data = json.loads(data.rstrip('\\n'))

print data['input']
print data['charybdis']
print data['spad-type']