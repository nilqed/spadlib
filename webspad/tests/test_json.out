{ "input":"D(x^n,x,4)","multiline?":"false","spad-type":"Expression(Integer)","algebra":"     4     3      2       n - 4
   (n  - 6n  + 11n  - 6n)x
","charybdis":"     4     3      2       n - 4
   (n  - 6n  + 11n  - 6n)x
                                                    Type: Expression(Integer)
","tex":"","html":"","mathml":"","formula":"","fortran":"","texmacs":"","openmath":"","format-flags": {"algebra":"true","tex":"false","html":"false","mathml":"false","formula":"false","fortran":"false","texmcas":"false","openmath":"false"}}

D(x^n,x,4)
     4     3      2       n - 4
   (n  - 6n  + 11n  - 6n)x
                                                    Type: Expression(Integer)

Expression(Integer)
